authors/apps/authentication/tests/test_registration.py

```diff
@@ -16,13 +20,86 @@ def test_user_register_successful(self):
                                     data=json.dumps(
                                         valid_register_data),
                                     content_type='application/json')
-
         self.assertEqual(response.status_code, status.HTTP_201_CREATED)
         self.assertEqual(response.data["email"],
                          valid_register_data["user"]["email"])
         self.assertEqual(response.data["username"],
                          valid_register_data["user"]["username"])
 
+    def test_activation_email_is_sent_on_valid_registration(self):
+        """
+        Test activation email is sent to user email upon
+        succesful registration.
+        """
+        self.register_test_user()
+        self.assertEqual(
+                    "Activation for your Author's Heaven account",
+                    mail.outbox[0].subject)
+        self.assertIn(
+                    "Thank you, Please Activate your account below.",
+                    mail.outbox[0].body)
+
+    def test_successful_account_activation(self):
+        """
+        Test when a user clicks on emailed activation link,
+        account is successfully acitvated.
+        """
+        self.register_test_user()
+        activation_link = (mail.outbox[0].body.split('\n')).pop(1)
+        url = activation_link.split("testserver").pop(1)
+        response = self.client.get(url, content_type='application/json')
+        self.assertEqual(
+                    response.data['msg'],
+                    "Your account is activated, enjoy")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_cannot_activate_with_expired_link(self):
+        """
+        Test user can not access activateion view with an expired token.
+        """
+        url = expired_link
+        response = self.client.get(url, content_type='application/json')
+        self.assertEqual(
+                    response.data['err_msg'], "Token expired")
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_cannot_activate_with_invalid_link(self):
```

Comment:
Hello @codjoero, I think you may have to start breaking these tests to create `unittests` and `integration` tests. The tests you've developed thus far do not show a clear distinction between the two. I'd recommend further reading of `Unittests vs integration` tests. You can start [here.](https://www.softwaretestingclass.com/what-is-difference-between-unit-testing-and-integration-testing/)

Reply:
Hello @malep2007, thanks for the feedback and resource. I think I understand the difference now. Would you like to see the different types in their separate folders? Thanks.